## THE BRIDGE (Data Science Bootcamp - Agosto 2020)
##  PROYECTO INDIVIDUAL
________________
### Análisis de datos sobre el clima en la ciudad de Madrid y en concreto a los datos registrados por la estación metereologica ubicada en el Parque del Retiro.

Fuente de datos: Agencia Estatal de Meteorologia (AEMET)   https://opendata.aemet.es/centrodedescargas/inicio

Autor: 
- José Luis Manjon  presimanjon@yahoo.es;        git @JoseLuisManjon
_________________

## 1. Librerías requeridas para el análisis (* Import *)
_________

In [2]:
import sys 
sys.path.append('../src/utils') 
import pandas as pd 
import json 
import requests
import seaborn as sns
import numpy as np
from datetime import datetime
import folders_tb as ftb
import visualization_tb as vtb
#import mining_data_tb as mtb
# visualization libraries
import matplotlib
import matplotlib.pyplot as plt
import seaborn as sns
import matplotlib.pyplot as plt
import plotly.graph_objs as go
import plotly.express as px 
import plotly.io as pio

# print the graphs in the notebook
%matplotlib inline

# set seaborn style to white
sns.set_style("white")

In [70]:
# url = "https://covid.ourworldindata.org/data/owid-covid-data.csv" 
df = pd.read_csv(url, encoding= 'unicode_escape',parse_dates=['y/m/d'], sep=",")
df

Carga de datos en un dataframe de un archivo csv descargado en el ordenador y que esta en la misma carpeta que el archivo VSC. En este caso son los datos del clima en Tokyo

In [3]:
# url = "https://covid.ourworldindata.org/data/owid-covid-data.csv" # Si el archivo esta descargado en el mismo directorio no es necesario y puedo suprimirlo
df = pd.read_csv('tokyo_data.csv', encoding= 'unicode_escape',parse_dates=['y/m/d'], sep=",")
df

y/m/d high(Ž) low(Ž) max_precipitation_10min(mm)  \
0        1984/1/1     10.1     0.9                           0   
1        1984/1/2      8.9     0.3                           0   
2        1984/1/3      7.7     0.9                           1   
3        1984/1/4     10.9     1.5                         0.5   
4        1984/1/5      7.6     1.9                           0   
...           ...      ...     ...                         ...   
12800  2018/12/27     13.2     4.9                           0   
12801  2018/12/28      8.4     1.8                           0   
12802  2018/12/29      9.8    -0.7                           0   
12803  2018/12/30     10.1     0.1                           0   
12804  2018/12/31      8.9     0.8                           0   

      daylight_hours(hours)  
0                       8.5  
1                         7  
2                       4.1  
3                       6.5  
4                       8.6  
...                     ...  
12800                     8  
12801                   8.9  
12802                   7.4  
12803                   8.8  
12804                   8.7  

[12805 rows x 5 columns]

## 2. Carga de datos inicial, de la fuente, en el dataframe        
## *madrid_clima*

Los datos se corresponden con los suministrados por las cuatro estaciones metereologicas que tiene la Agencia Estatal de Metereología en la ciudad de Madrid. Los datos se descargan desde la pagina web de la AEMET en archivos json. 
El estudio se basa en los datos diarios de esas estaciones desde el año 1980 hasta finales del 2019. 
____________

Ahora descargo los datos de las estaciónes metereologicas en la ciudad de Madrid que suministra la AEMET en archivos json. AEMET ha limitado el tamaño del archivo json que se puede descargar a menos de cinco años, por lo que ha sido necesario repetir el proceso varias veces hasta abarcar el periodo en estudio. 

In [71]:
#url1 = 'https://opendata.aemet.es/opendata/sh/7289e425/?api_key=%20eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJwcmVzaW1hbmpvbkB5YWhvby5lcyIsImp0aSI6IjE2NGZlNTUzLWNhMTctNGExYy1iNmM5LTExY2UyYTVhM2UzOCIsImlzcyI6IkFFTUVUIiwiaWF0IjoxNTk3NDg5MDAwLCJ1c2VySWQiOiIxNjRmZTU1My1jYTE3LTRhMWMtYjZjOS0xMWNlMmE1YTNlMzgiLCJyb2xlIjoiIn0.nx__pRBlaMxfCkGE5c5SdIIOc5xYM--ZtXERecvkPWw%27'

In [72]:
#m1 = requests.get(url1)
#json_readed1 = json.loads(m1.text)


Save with indent

In [73]:
#with open('data_indented_madrid1.json', 'w+') as outfile:
    #json.dump(json_readed1, outfile, indent=4)

Creo el dataframe

In [4]:
df_madrid1 = pd.read_json("data_indented_madrid1.json")
df_madrid1.head()

fecha indicativo                        nombre provincia  altitud  \
0  1980-01-01       3129             MADRID AEROPUERTO    MADRID      609   
1  1980-01-01      3194U  MADRID, CIUDAD UNIVERSITARIA    MADRID      664   
2  1980-01-01       3196        MADRID, CUATRO VIENTOS    MADRID      690   
3  1980-01-01       3195                MADRID, RETIRO    MADRID      667   
4  1980-01-02       3129             MADRID AEROPUERTO    MADRID      609   

   tmed prec  tmin horatmin  tmax horatmax   dir velmedia racha horaracha  \
0  13,6  0,0  11,0   Varias  16,1    14:30  29.0      6,9  13,3    Varias   
1  12,6  0,0  10,0   Varias  15,3    15:20   NaN      5,0   NaN       NaN   
2  11,8  0,0   8,4    04:40  15,2    13:20  27.0      5,3  11,1     16:20   
3  12,5  0,0   9,7    06:15  15,3    13:30  29.0      5,3  11,1     11:20   
4   7,7  0,0   3,4    06:00  12,0    14:00  27.0      1,9   8,9     00:00   

   sol presMax horaPresMax presMin horaPresMin  
0  7,9   950,9          23   948,7      Varias  
1  8,9   942,1          24   939,7          05  
2  7,7   939,7          24   937,6          06  
3  8,2     NaN         NaN     NaN         NaN  
4  3,8   957,6          23   950,9      Varias

In [8]:
#url2 = 'https://opendata.aemet.es/opendata/sh/d465d8d0/?api_key=eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJwcmVzaW1hbmpvbkB5YWhvby5lcyIsImp0aSI6Ijk2MmI2NzlhLThjODUtNGFmZC04NWEzLTQzMWZiN2MzZTJlZCIsImlzcyI6IkFFTUVUIiwiaWF0IjoxNTk3OTE3ODg0LCJ1c2VySWQiOiI5NjJiNjc5YS04Yzg1LTRhZmQtODVhMy00MzFmYjdjM2UyZWQiLCJyb2xlIjoiIn0.SLgACuFBSGU7eoljv3Aqtp0d5Xmdl-00uf-bV7m7aRA'

In [9]:
#m2 = requests.get(url2)
#json_readed2 = json.loads(m2.text)


In [10]:
#with open('data_indented_madrid2.json', 'w+') as outfile:
    #json.dump(json_readed2, outfile, indent=4)

In [5]:
df_madrid2 = pd.read_json("data_indented_madrid2.json")
df_madrid2.tail()

fecha indicativo                  nombre provincia  altitud  tmed  \
5686  1989-10-31       3196  MADRID, CUATRO VIENTOS    MADRID      690  16,7   
5687  1989-10-31       3195          MADRID, RETIRO    MADRID      667  16,4   
5688  1989-11-01       3129       MADRID AEROPUERTO    MADRID      609  13,2   
5689  1989-11-01       3196  MADRID, CUATRO VIENTOS    MADRID      690  16,3   
5690  1989-11-01       3195          MADRID, RETIRO    MADRID      667  16,6   

     prec  tmin horatmin  tmax horatmax   dir velmedia racha horaracha  sol  \
5686  0,0   9,2    07:50  24,2    15:50  17.0      0,8   2,5     11:58  7,5   
5687  0,0  10,2    07:00  22,6    13:00   4.0      1,1   4,2     23:50  8,9   
5688  0,0   2,5    06:00  24,0    15:00  25.0      2,5   7,2     15:40  9,4   
5689  0,0   9,0    07:00  23,6    15:15  21.0      0,3   6,1     14:47  9,4   
5690  0,0  10,8    07:00  22,3    16:00  22.0      1,1   6,9     14:56  8,8   

     presMax horaPresMax presMin horaPresMin  
5686   948,2          10   944,5          24  
5687   951,9          09   948,1          24  
5688   956,9      Varias   952,6          24  
5689   944,5          00   940,3          17  
5690   948,1          00   944,2          17

In [12]:
#url3 = 'https://opendata.aemet.es/opendata/sh/2932e4d4/?api_key=eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJwcmVzaW1hbmpvbkB5YWhvby5lcyIsImp0aSI6Ijk2MmI2NzlhLThjODUtNGFmZC04NWEzLTQzMWZiN2MzZTJlZCIsImlzcyI6IkFFTUVUIiwiaWF0IjoxNTk3OTE3ODg0LCJ1c2VySWQiOiI5NjJiNjc5YS04Yzg1LTRhZmQtODVhMy00MzFmYjdjM2UyZWQiLCJyb2xlIjoiIn0.SLgACuFBSGU7eoljv3Aqtp0d5Xmdl-00uf-bV7m7aRA'

In [13]:
#m3 = requests.get(url3)
#json_readed3 = json.loads(m3.text)

In [14]:
#with open('data_indented_madrid3.json', 'w+') as outfile:
    #json.dump(json_readed3, outfile, indent=4)

In [6]:
df_madrid3 = pd.read_json("data_indented_madrid3.json")
df_madrid3.tail()

fecha  indicativo                  nombre provincia  altitud  tmed  \
5380  1994-09-30        3196  MADRID, CUATRO VIENTOS    MADRID      690  17,5   
5381  1994-09-30        3195          MADRID, RETIRO    MADRID      667  17,8   
5382  1994-10-01        3129       MADRID AEROPUERTO    MADRID      609  18,6   
5383  1994-10-01        3196  MADRID, CUATRO VIENTOS    MADRID      690  18,2   
5384  1994-10-01        3195          MADRID, RETIRO    MADRID      667  18,4   

     prec  tmin horatmin  tmax horatmax   dir velmedia racha horaracha  sol  \
5380   Ip  12,4    05:00  22,6    15:40  18.0      1,1   3,9     13:35  5,1   
5381  0,0  14,5    03:10  21,1    13:40   5.0      1,4   5,8     00:30  5,3   
5382   Ip  12,4    05:00  24,8    15:30  34.0      2,2   9,4     15:15  5,2   
5383   Ip  13,0    06:00  23,4    13:40  14.0      2,2   6,1     09:45  4,5   
5384  2,4  13,8    23:00  23,0    13:40  33.0      1,9   8,9     15:20  5,9   

     presMax horaPresMax presMin horaPresMin  
5380   939,2          24   935,5          05  
5381   939,7          24   935,8      Varias  
5382   953,5          24   950,3          15  
5383   941,5          22   938,7          15  
5384   942,1      Varias   939,1          04

In [16]:
#url4 = 'https://opendata.aemet.es/opendata/sh/f1545629/?api_key=eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJwcmVzaW1hbmpvbkB5YWhvby5lcyIsImp0aSI6Ijk2MmI2NzlhLThjODUtNGFmZC04NWEzLTQzMWZiN2MzZTJlZCIsImlzcyI6IkFFTUVUIiwiaWF0IjoxNTk3OTE3ODg0LCJ1c2VySWQiOiI5NjJiNjc5YS04Yzg1LTRhZmQtODVhMy00MzFmYjdjM2UyZWQiLCJyb2xlIjoiIn0.SLgACuFBSGU7eoljv3Aqtp0d5Xmdl-00uf-bV7m7aRA'

In [17]:
#m4 = requests.get(url4)
#json_readed4 = json.loads(m4.text)

In [18]:
#with open('data_indented_madrid4.json', 'w+') as outfile:
    #json.dump(json_readed4, outfile, indent=4)

In [7]:
df_madrid4 = pd.read_json("data_indented_madrid4.json")
df_madrid4.tail(3)

fecha indicativo                        nombre provincia  altitud  \
6120  1999-09-01      3194U  MADRID, CIUDAD UNIVERSITARIA    MADRID      664   
6121  1999-09-01       3196        MADRID, CUATRO VIENTOS    MADRID      690   
6122  1999-09-01       3195                MADRID, RETIRO    MADRID      667   

      tmed  prec  tmin horatmin  tmax horatmax   dir velmedia racha horaracha  \
6120  24,8  10,7  19,6    23:59  29,9   Varias  10.0      2,2  11,7     20:20   
6121  24,8  15,9  19,6   Varias  30,0   Varias  16.0      3,9  12,8     13:29   
6122  23,8   6,5  19,0    23:50  28,7    13:15  15.0      2,8  11,9     13:30   

      sol presMax horaPresMax presMin horaPresMin  
6120  NaN     NaN         NaN     NaN         NaN  
6121  6,9   938,7          24   936,5          15  
6122  6,4   940,3          24   937,4          15

In [20]:
#url5 = 'https://opendata.aemet.es/opendata/sh/96b7b5ac/?api_key=eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJwcmVzaW1hbmpvbkB5YWhvby5lcyIsImp0aSI6IjE2NGZlNTUzLWNhMTctNGExYy1iNmM5LTExY2UyYTVhM2UzOCIsImlzcyI6IkFFTUVUIiwiaWF0IjoxNTk3NDg5MDAwLCJ1c2VySWQiOiIxNjRmZTU1My1jYTE3LTRhMWMtYjZjOS0xMWNlMmE1YTNlMzgiLCJyb2xlIjoiIn0.nx__pRBlaMxfCkGE5c5SdIIOc5xYM--ZtXERecvkPWw'

In [21]:
#m5 = requests.get(url5)
#json_readed5 = json.loads(m5.text)

In [22]:
#with open('data_indented_madrid5.json', 'w+') as outfile:
    #json.dump(json_readed5, outfile, indent=4)

In [8]:
df_madrid5 = pd.read_json("data_indented_madrid5.json")
df_madrid5.tail(3)

fecha indicativo                        nombre provincia  altitud  \
6949  2004-08-01      3194U  MADRID, CIUDAD UNIVERSITARIA    MADRID      664   
6950  2004-08-01       3196        MADRID, CUATRO VIENTOS    MADRID      690   
6951  2004-08-01       3195                MADRID, RETIRO    MADRID      667   

      tmed prec  tmin horatmin  tmax horatmax   dir velmedia racha horaracha  \
6949  29,9  0,0  23,4    05:35  36,4    16:30  15.0      0,8   8,9     18:50   
6950  29,8   Ip  23,2   Varias  36,4    16:30  14.0      2,5  10,8     18:32   
6951  28,9  0,2  23,2    05:50  34,6    14:10  17.0      2,2  13,1     18:40   

      sol presMax horaPresMax presMin horaPresMin  
6949  NaN     NaN         NaN     NaN         NaN  
6950  8,6   940,6      Varias   936,6          18  
6951  7,7   941,9          07   937,8          18

In [24]:
#url6 = 'https://opendata.aemet.es/opendata/sh/a3c2e740/?api_key=eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJwcmVzaW1hbmpvbkB5YWhvby5lcyIsImp0aSI6IjE2NGZlNTUzLWNhMTctNGExYy1iNmM5LTExY2UyYTVhM2UzOCIsImlzcyI6IkFFTUVUIiwiaWF0IjoxNTk3NDg5MDAwLCJ1c2VySWQiOiIxNjRmZTU1My1jYTE3LTRhMWMtYjZjOS0xMWNlMmE1YTNlMzgiLCJyb2xlIjoiIn0.nx__pRBlaMxfCkGE5c5SdIIOc5xYM--ZtXERecvkPWw'

In [25]:
#m6 = requests.get(url6)
#json_readed6 = json.loads(m6.text)
#json_readed6

In [26]:
#with open('data_indented_madrid6.json', 'w+') as outfile:
    #json.dump(json_readed6, outfile, indent=4)

In [9]:
df_madrid6 = pd.read_json("data_indented_madrid6.json")
df_madrid6.tail(3)

fecha indicativo                        nombre provincia  altitud  \
6757  2009-07-01      3194U  MADRID, CIUDAD UNIVERSITARIA    MADRID      664   
6758  2009-07-01       3196        MADRID, CUATRO VIENTOS    MADRID      690   
6759  2009-07-01       3195                MADRID, RETIRO    MADRID      667   

      tmed prec  tmin horatmin  tmax horatmax   dir velmedia racha horaracha  \
6757  27,5  0,0  19,0    05:35  36,0   Varias  26.0      2,5  13,1     14:10   
6758  28,2  0,0  21,4    05:00  35,0    16:00  26.0      3,9  11,9     13:50   
6759  28,6  0,0  21,6    04:30  35,5    14:15  20.0      3,1  10,6     16:20   

       sol presMax horaPresMax presMin horaPresMin  
6757   NaN     NaN         NaN     NaN         NaN  
6758  13,4   941,0          07   937,8          17  
6759   NaN   943,7          10   940,5          18

Para tener mas datos y poder compararlos con Madrid, descargo todos los datos de todas las estaciones de España durante el mes de Enero del 2000.

Esto es solo una idea, no la he llevado a la practica todavia.

In [28]:
#url7 = 'https://opendata.aemet.es/opendata/sh/6ed3fc2e/?api_key=eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJwcmVzaW1hbmpvbkB5YWhvby5lcyIsImp0aSI6IjE2NGZlNTUzLWNhMTctNGExYy1iNmM5LTExY2UyYTVhM2UzOCIsImlzcyI6IkFFTUVUIiwiaWF0IjoxNTk3NDg5MDAwLCJ1c2VySWQiOiIxNjRmZTU1My1jYTE3LTRhMWMtYjZjOS0xMWNlMmE1YTNlMzgiLCJyb2xlIjoiIn0.nx__pRBlaMxfCkGE5c5SdIIOc5xYM--ZtXERecvkPWw'

In [29]:
#m7 = requests.get(url7)
#json_readed7 = json.loads(m7.text)
#json_readed_prueba

In [30]:
#with open('data_indented_madrid7.json', 'w+') as outfile:
    #json.dump(json_readed7, outfile, indent=4)

In [10]:
df_madrid7 = pd.read_json("data_indented_madrid7.json")
df_madrid7.tail(4)

fecha indicativo                        nombre provincia  altitud  \
7082  2014-06-01       3129             MADRID AEROPUERTO    MADRID      609   
7083  2014-06-01      3194U  MADRID, CIUDAD UNIVERSITARIA    MADRID      664   
7084  2014-06-01       3196        MADRID, CUATRO VIENTOS    MADRID      690   
7085  2014-06-01       3195                MADRID, RETIRO    MADRID      667   

      tmed prec  tmin horatmin  tmax horatmax   dir velmedia racha horaracha  \
7082  18,7  0,0  11,3    04:10  26,1    13:54  36.0      3,9  11,4     19:55   
7083  18,2  0,0  11,5    05:00  24,8    15:00   NaN      NaN   NaN       NaN   
7084  18,0  0,0  11,5    04:30  24,5    14:30   3.0      3,1   9,7     00:10   
7085  18,1  0,0  11,6    04:50  24,6    15:50   2.0      2,5  12,2     07:20   

       sol presMax horaPresMax presMin horaPresMin  
7082   9,4   954,0          00   950,1          18  
7083   NaN     NaN         NaN     NaN         NaN  
7084  11,0   942,0          00   938,8      Varias  
7085   NaN   944,1          00   939,9          19

In [32]:
#url8 = 'https://opendata.aemet.es/opendata/sh/6e556a02/?api_key=eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJwcmVzaW1hbmpvbkB5YWhvby5lcyIsImp0aSI6IjE2NGZlNTUzLWNhMTctNGExYy1iNmM5LTExY2UyYTVhM2UzOCIsImlzcyI6IkFFTUVUIiwiaWF0IjoxNTk3NDg5MDAwLCJ1c2VySWQiOiIxNjRmZTU1My1jYTE3LTRhMWMtYjZjOS0xMWNlMmE1YTNlMzgiLCJyb2xlIjoiIn0.nx__pRBlaMxfCkGE5c5SdIIOc5xYM--ZtXERecvkPWw'

In [33]:
#m8 = requests.get(url8)
#json_readed8 = json.loads(m8.text)

In [34]:
#with open('data_indented_madrid8.json', 'w+') as outfile:
    #json.dump(json_readed8, outfile, indent=4)

In [11]:
df_madrid8 = pd.read_json("data_indented_madrid8.json")
df_madrid8.tail(4)

fecha indicativo                        nombre provincia  altitud  \
6616  2018-12-31       3129             MADRID AEROPUERTO    MADRID      609   
6617  2018-12-31      3194U  MADRID, CIUDAD UNIVERSITARIA    MADRID      664   
6618  2018-12-31       3196        MADRID, CUATRO VIENTOS    MADRID      690   
6619  2018-12-31       3195                MADRID, RETIRO    MADRID      667   

     tmed prec  tmin horatmin  tmax horatmax   dir velmedia racha horaracha  \
6616  6,2  0,0  -3,7    07:50  16,1    15:34  99.0      0,8   3,6    Varias   
6617  6,6  0,0  -3,0    08:00  16,3    16:00  24.0      0,3   3,1     13:20   
6618  8,2  0,0   0,4    08:00  16,0    15:00  35.0      1,4   4,2     01:40   
6619  7,0  0,0   0,7    07:50  13,4    15:30   6.0      0,6   5,3     23:59   

      sol presMax horaPresMax presMin horaPresMin  
6616  9,0   964,1      Varias   961,0          15  
6617  NaN     NaN         NaN     NaN         NaN  
6618  9,0   951,6      Varias   949,1          16  
6619  NaN   954,0          10   951,6          15

In [36]:
#url9 = 'https://opendata.aemet.es/opendata/sh/450be29f/?api_key=eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJwcmVzaW1hbmpvbkB5YWhvby5lcyIsImp0aSI6Ijk2MmI2NzlhLThjODUtNGFmZC04NWEzLTQzMWZiN2MzZTJlZCIsImlzcyI6IkFFTUVUIiwiaWF0IjoxNTk3OTE3ODg0LCJ1c2VySWQiOiI5NjJiNjc5YS04Yzg1LTRhZmQtODVhMy00MzFmYjdjM2UyZWQiLCJyb2xlIjoiIn0.SLgACuFBSGU7eoljv3Aqtp0d5Xmdl-00uf-bV7m7aRA'

In [37]:
#m9 = requests.get(url9)
#json_readed9 = json.loads(m9.text)

In [38]:
#with open('data_indented_madrid9.json', 'w+') as outfile:
    #json.dump(json_readed9, outfile, indent=4)

In [12]:
df_madrid9 = pd.read_json("data_indented_madrid9.json")
df_madrid9.tail(4)

fecha indicativo                        nombre provincia  altitud  \
1456  2019-12-31       3129             MADRID AEROPUERTO    MADRID      609   
1457  2019-12-31      3194U  MADRID, CIUDAD UNIVERSITARIA    MADRID      664   
1458  2019-12-31       3196        MADRID, CUATRO VIENTOS    MADRID      690   
1459  2019-12-31       3195                MADRID, RETIRO    MADRID      667   

     tmed prec  tmin horatmin  tmax horatmax   dir velmedia racha horaracha  \
1456  5,8  0,0  -1,4    07:05  12,9    14:28  17.0      0,6   4,7     14:48   
1457  6,0  0,0  -1,5    07:50  13,4    15:00   NaN      NaN   NaN       NaN   
1458  7,3  0,0   1,5    07:30  13,1   Varias  35.0      0,8   3,1     02:01   
1459  6,8  0,0   2,3    07:30  11,4    16:00   7.0      0,3   3,1     09:30   

      sol presMax horaPresMax presMin horaPresMin  
1456  8,3   964,4      Varias   961,7          14  
1457  NaN     NaN         NaN     NaN         NaN  
1458  8,7   952,0      Varias   949,7      Varias  
1459  NaN   954,2          10   951,8      Varias

In [40]:
#url25 = 'https://opendata.aemet.es/opendata/sh/c9d5dd53/?api_key=eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJwcmVzaW1hbmpvbkB5YWhvby5lcyIsImp0aSI6Ijk2MmI2NzlhLThjODUtNGFmZC04NWEzLTQzMWZiN2MzZTJlZCIsImlzcyI6IkFFTUVUIiwiaWF0IjoxNTk3OTE3ODg0LCJ1c2VySWQiOiI5NjJiNjc5YS04Yzg1LTRhZmQtODVhMy00MzFmYjdjM2UyZWQiLCJyb2xlIjoiIn0.SLgACuFBSGU7eoljv3Aqtp0d5Xmdl-00uf-bV7m7aRA'

In [41]:
#m25 = requests.get(url25)
#json_readed25 = json.loads(m25.text)

In [42]:
#with open('data_indented_madrid25.json', 'w+') as outfile:
    #json.dump(json_readed25, outfile, indent=4)

In [22]:
df_madrid25 = pd.read_json("data_indented_madrid25.json")
df_madrid25.head(4)

fecha  indicativo          nombre provincia  altitud tmed prec tmin  \
0  1925-01-01        3195  MADRID, RETIRO    MADRID      667  4,2   Ip  2,4   
1  1925-01-02        3195  MADRID, RETIRO    MADRID      667  5,6  0,0  4,4   
2  1925-01-03        3195  MADRID, RETIRO    MADRID      667  4,8   Ip  3,2   
3  1925-01-04        3195  MADRID, RETIRO    MADRID      667  7,0  0,0  4,8   

  horatmin tmax horatmax   dir velmedia racha horaracha  sol presMax  \
0    06:00  6,0    15:00  99.0      0,3   1,9    Varias  0,0   951,9   
1    07:15  6,8    15:00  20.0      0,0   1,4     10:15  0,0   951,4   
2    08:00  6,4    16:30  16.0      0,6   3,1     04:00  0,1   951,1   
3    08:30  9,2    15:00  16.0      0,3   2,5     09:20  0,0   951,8   

  horaPresMax presMin horaPresMin  
0          10   949,8          01  
1          11   949,7          06  
2          00   948,5          15  
3          24   948,3          06

In [44]:
df_madrid9.index

RangeIndex(start=0, stop=1460, step=1)

In [45]:
df_madrid9.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1460 entries, 0 to 1459
Data columns (total 20 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   fecha        1460 non-null   object 
 1   indicativo   1460 non-null   object 
 2   nombre       1460 non-null   object 
 3   provincia    1460 non-null   object 
 4   altitud      1460 non-null   int64  
 5   tmed         1405 non-null   object 
 6   prec         1458 non-null   object 
 7   tmin         1405 non-null   object 
 8   horatmin     1404 non-null   object 
 9   tmax         1405 non-null   object 
 10  horatmax     1405 non-null   object 
 11  dir          1388 non-null   float64
 12  velmedia     1391 non-null   object 
 13  racha        1388 non-null   object 
 14  horaracha    1388 non-null   object 
 15  sol          729 non-null    object 
 16  presMax      1095 non-null   object 
 17  horaPresMax  1095 non-null   object 
 18  presMin      1095 non-null   object 
 19  horaPr

In [76]:
df_madrid1.shape, df_madrid2.shape, df_madrid3.shape, df_madrid4.shape, df_madrid5.shape, df_madrid6.shape, df_madrid7.shape, df_madrid8.shape, df_madrid9.shape


((7188, 20),
 (5691, 20),
 (5385, 20),
 (6123, 20),
 (6952, 20),
 (6760, 20),
 (7086, 20),
 (6620, 20),
 (1460, 20))

## 3. Preparación de los datos - Data wrangling and Data cleaning
________________
Exploración de los datos del dataframe

Uno todos los df descargados en uno y lo llamo madrid_8019

In [13]:
madrid_8019 = pd.concat([df_madrid1, df_madrid2, df_madrid3, df_madrid4, df_madrid5, df_madrid6, df_madrid7, df_madrid8, df_madrid9,], axis=0)
madrid_8019.head(4)

fecha indicativo                        nombre provincia  altitud  \
0  1980-01-01       3129             MADRID AEROPUERTO    MADRID      609   
1  1980-01-01      3194U  MADRID, CIUDAD UNIVERSITARIA    MADRID      664   
2  1980-01-01       3196        MADRID, CUATRO VIENTOS    MADRID      690   
3  1980-01-01       3195                MADRID, RETIRO    MADRID      667   

   tmed prec  tmin horatmin  tmax horatmax   dir velmedia racha horaracha  \
0  13,6  0,0  11,0   Varias  16,1    14:30  29.0      6,9  13,3    Varias   
1  12,6  0,0  10,0   Varias  15,3    15:20   NaN      5,0   NaN       NaN   
2  11,8  0,0   8,4    04:40  15,2    13:20  27.0      5,3  11,1     16:20   
3  12,5  0,0   9,7    06:15  15,3    13:30  29.0      5,3  11,1     11:20   

   sol presMax horaPresMax presMin horaPresMin  
0  7,9   950,9          23   948,7      Varias  
1  8,9   942,1          24   939,7          05  
2  7,7   939,7          24   937,6          06  
3  8,2     NaN         NaN     NaN         NaN

In [96]:
# Numero de NaN en el dataframe por columnas
madrid_8019.isnull().sum() 

fecha              0
indicativo         0
nombre             0
provincia          0
altitud            0
tmed             323
prec             447
tmin             323
horatmin         347
tmax             321
horatmax         331
dir             2693
velmedia        1974
racha           2784
horaracha       2835
sol            13257
presMax         7431
horaPresMax     7431
presMin         7431
horaPresMin     7440
dtype: int64

In [97]:
# Veo la cantidad de NaN que hay en la tabla
madrid_8019.isnull().sum().sum() 

55368

In [14]:
# Vemos cuantos registros hay para cada estación de medida, la cantidad y la ubicación de los Nan
m_aeropuerto = madrid_8019[madrid_8019['nombre']== 'MADRID AEROPUERTO']
m_aeropuerto.shape

(14610, 20)

In [51]:
m_aeropuerto.isnull().sum().sum() 

220

In [52]:
m_aeropuerto.isnull().sum()

fecha           0
indicativo      0
nombre          0
provincia       0
altitud         0
tmed            6
prec            1
tmin            6
horatmin        6
tmax            6
horatmax        6
dir            24
velmedia        6
racha          24
horaracha      75
sol            48
presMax         3
horaPresMax     3
presMin         3
horaPresMin     3
dtype: int64

In [15]:
m_retiro = madrid_8019[madrid_8019['nombre']== 'MADRID, RETIRO']
m_retiro.shape

(14610, 20)

In [54]:
m_retiro.isnull().sum().sum() 

8379

In [55]:
m_retiro.isnull().sum()

fecha             0
indicativo        0
nombre            0
provincia         0
altitud           0
tmed              3
prec              0
tmin              3
horatmin          3
tmax              3
horatmax          5
dir             527
velmedia        503
racha           527
horaracha       602
sol            5857
presMax          87
horaPresMax      87
presMin          86
horaPresMin      86
dtype: int64

In [16]:
m_cuatrovientos = madrid_8019[madrid_8019['nombre']== 'MADRID, CUATRO VIENTOS']
m_cuatrovientos.shape

(14610, 20)

In [57]:
m_cuatrovientos.isnull().sum().sum() 

237

In [58]:
m_cuatrovientos.isnull().sum()

fecha            0
indicativo       0
nombre           0
provincia        0
altitud          0
tmed             2
prec             0
tmin             2
horatmin        14
tmax             0
horatmax         1
dir             21
velmedia        18
racha          111
horaracha       26
sol             13
presMax          5
horaPresMax      5
presMin          5
horaPresMin     14
dtype: int64

In [17]:
m_universitaria = madrid_8019[madrid_8019['nombre']== 'MADRID, CIUDAD UNIVERSITARIA']
m_universitaria.shape

(9435, 20)

In [60]:
m_universitaria.isnull().sum().sum() 

46525

In [61]:
m_universitaria.isnull().sum()

fecha             0
indicativo        0
nombre            0
provincia         0
altitud           0
tmed            312
prec            446
tmin            312
horatmin        324
tmax            312
horatmax        319
dir            2120
velmedia       1447
racha          2121
horaracha      2131
sol            7339
presMax        7335
horaPresMax    7335
presMin        7336
horaPresMin    7336
dtype: int64

In [18]:
# Revisión de los datos en la Ciudad Universitaria debido al alto número de valores Nan que presenta.
m_universitaria[(m_universitaria.tmed.isnull()) | (m_universitaria.tmin.isnull())]

fecha indicativo                        nombre provincia  altitud  \
2551  1997-01-22      3194U  MADRID, CIUDAD UNIVERSITARIA    MADRID      664   
2562  1997-01-25      3194U  MADRID, CIUDAD UNIVERSITARIA    MADRID      664   
2693  1997-03-03      3194U  MADRID, CIUDAD UNIVERSITARIA    MADRID      664   
2771  1997-03-23      3194U  MADRID, CIUDAD UNIVERSITARIA    MADRID      664   
2966  1997-05-11      3194U  MADRID, CIUDAD UNIVERSITARIA    MADRID      664   
...          ...        ...                           ...       ...      ...   
717   2019-06-29      3194U  MADRID, CIUDAD UNIVERSITARIA    MADRID      664   
721   2019-06-30      3194U  MADRID, CIUDAD UNIVERSITARIA    MADRID      664   
725   2019-07-01      3194U  MADRID, CIUDAD UNIVERSITARIA    MADRID      664   
729   2019-07-02      3194U  MADRID, CIUDAD UNIVERSITARIA    MADRID      664   
1265  2019-11-13      3194U  MADRID, CIUDAD UNIVERSITARIA    MADRID      664   

     tmed prec tmin horatmin tmax horatmax   dir velmedia racha horaracha  \
2551  NaN  NaN  NaN      NaN  NaN      NaN   NaN      2,2   NaN       NaN   
2562  NaN  NaN  NaN      NaN  NaN      NaN   NaN      1,1   NaN       NaN   
2693  NaN  0,0  NaN      NaN  NaN      NaN   NaN      1,1   NaN       NaN   
2771  NaN  NaN  NaN      NaN  NaN      NaN   NaN      1,7   NaN       NaN   
2966  NaN  0,0  NaN      NaN  NaN      NaN   NaN      1,9   NaN       NaN   
...   ...  ...  ...      ...  ...      ...   ...      ...   ...       ...   
717   NaN  0,0  NaN      NaN  NaN      NaN  12.0      1,1   8,1     22:50   
721   NaN  0,0  NaN      NaN  NaN      NaN  27.0      1,7   8,3     14:40   
725   NaN  0,0  NaN      NaN  NaN      NaN   7.0      2,5  12,5     20:20   
729   NaN  0,0  NaN      NaN  NaN      NaN   9.0      1,1   9,4     23:40   
1265  NaN  1,5  NaN      NaN  NaN      NaN   NaN      NaN   NaN       NaN   

      sol presMax horaPresMax presMin horaPresMin  
2551  NaN     NaN         NaN     NaN         NaN  
2562  NaN     NaN         NaN     NaN         NaN  
2693  NaN     NaN         NaN     NaN         NaN  
2771  NaN     NaN         NaN     NaN         NaN  
2966  NaN     NaN         NaN     NaN         NaN  
...   ...     ...         ...     ...         ...  
717   NaN     NaN         NaN     NaN         NaN  
721   NaN     NaN         NaN     NaN         NaN  
725   NaN     NaN         NaN     NaN         NaN  
729   NaN     NaN         NaN     NaN         NaN  
1265  NaN     NaN         NaN     NaN         NaN  

[312 rows x 20 columns]

Tras analizar los datos existentes en cada estación se comprueba que en la estación metereológica de la Ciudad Universitaria no se han guardado registros durante 5175 dias, a lo que hay que sumar los días en los que los valores registrados son nulos, 312 en las temperaturas  y 446 en cuanto a la precipitación. Esto indica que en el perioro 1980 - 2019 no dispondriamos de registros climáticos durante mas de 15 años. Por ello se toma la decisión de no utilizar los resgistros existentes de la Ciudad Universitaria para analizar el clima en Madrid.

In [92]:
# Eliminamos del df madrid_8019 los registros correspondientes a la Ciudad Universitaria
madrid_clima3 =  madrid_8019[~(madrid_8019['nombre']=='MADRID, CIUDAD UNIVERSITARIA')]
madrid_clima3

fecha indicativo                  nombre provincia  altitud  tmed  \
0     1980-01-01       3129       MADRID AEROPUERTO    MADRID      609  13,6   
2     1980-01-01       3196  MADRID, CUATRO VIENTOS    MADRID      690  11,8   
3     1980-01-01       3195          MADRID, RETIRO    MADRID      667  12,5   
4     1980-01-02       3129       MADRID AEROPUERTO    MADRID      609   7,7   
6     1980-01-02       3196  MADRID, CUATRO VIENTOS    MADRID      690   8,5   
...          ...        ...                     ...       ...      ...   ...   
1454  2019-12-30       3196  MADRID, CUATRO VIENTOS    MADRID      690   7,4   
1455  2019-12-30       3195          MADRID, RETIRO    MADRID      667   7,0   
1456  2019-12-31       3129       MADRID AEROPUERTO    MADRID      609   5,8   
1458  2019-12-31       3196  MADRID, CUATRO VIENTOS    MADRID      690   7,3   
1459  2019-12-31       3195          MADRID, RETIRO    MADRID      667   6,8   

     prec  tmin horatmin  tmax horatmax   dir velmedia racha horaracha  sol  \
0     0,0  11,0   Varias  16,1    14:30  29.0      6,9  13,3    Varias  7,9   
2     0,0   8,4    04:40  15,2    13:20  27.0      5,3  11,1     16:20  7,7   
3     0,0   9,7    06:15  15,3    13:30  29.0      5,3  11,1     11:20  8,2   
4     0,0   3,4    06:00  12,0    14:00  27.0      1,9   8,9     00:00  3,8   
6     0,0   4,0    23:59  13,0    14:40  29.0      1,1  11,1     04:00  6,6   
...   ...   ...      ...   ...      ...   ...      ...   ...       ...  ...   
1454  0,0   1,5    05:10  13,2    14:10  14.0      0,6   4,2     11:30  8,9   
1455  0,0   2,5    05:30  11,5   Varias   NaN      0,8   NaN       NaN  NaN   
1456  0,0  -1,4    07:05  12,9    14:28  17.0      0,6   4,7     14:48  8,3   
1458  0,0   1,5    07:30  13,1   Varias  35.0      0,8   3,1     02:01  8,7   
1459  0,0   2,3    07:30  11,4    16:00   7.0      0,3   3,1     09:30  NaN   

     presMax horaPresMax presMin horaPresMin  
0      950,9          23   948,7      Varias  
2      939,7          24   937,6          06  
3        NaN         NaN     NaN         NaN  
4      957,6          23   950,9      Varias  
6      944,7          24   939,7          00  
...      ...         ...     ...         ...  
1454   951,9          11   949,5      Varias  
1455   954,0          11   950,9          00  
1456   964,4      Varias   961,7          14  
1458   952,0      Varias   949,7      Varias  
1459   954,2          10   951,8      Varias  

[43830 rows x 20 columns]

Analizando la distribución de los valores Nan en el dataframe madrid_clima nos encontramos que en las columnas relativas a la temperatura y la precipitación su número no es significativo por lo que rellenaremos estos registros interpolando entre el valor anterior y el posterior, por ser la manera que menor influencia va a tener para el estudio. Para ello, en primer lugar cambiaremos los datos de type string a float (cambiando la coma del separador de los decmiales por el punto) y despues actuaremos sobre los Nan de esas columanas. Y por último creremos una nueva estación metereológica que recogerá los valores promedio de las otras tres.

In [93]:
madrid_clima3.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 43830 entries, 0 to 1459
Data columns (total 20 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   fecha        43830 non-null  object 
 1   indicativo   43830 non-null  object 
 2   nombre       43830 non-null  object 
 3   provincia    43830 non-null  object 
 4   altitud      43830 non-null  int64  
 5   tmed         43819 non-null  object 
 6   prec         43829 non-null  object 
 7   tmin         43819 non-null  object 
 8   horatmin     43807 non-null  object 
 9   tmax         43821 non-null  object 
 10  horatmax     43818 non-null  object 
 11  dir          43258 non-null  float64
 12  velmedia     43303 non-null  object 
 13  racha        43168 non-null  object 
 14  horaracha    43127 non-null  object 
 15  sol          37912 non-null  object 
 16  presMax      43734 non-null  object 
 17  horaPresMax  43734 non-null  object 
 18  presMin      43735 non-null  object 
 19  horaP

In [94]:
# Sustituyo la coma por el punto como separación decimal en los valores tipo string
col_convert = ['tmed','prec','tmin','tmax','velmedia','racha','sol','presMax','presMin']
for col in col_convert:
    madrid_clima3[col] = madrid_clima3[col].str.replace(',','.')

In [95]:
#Los transformo de tipo string a tipo float
#col_convert = ['tmed','prec','tmin','tmax', 'velmedia','racha','sol','presMax','presMin']
for col in col_convert:
    madrid_clima3[col] = pd.to_numeric(madrid_clima3[col], errors='coerce',downcast='float')

In [96]:
# Sustituyo los valores Nan en las siguientes columnas interpolando entre el valor anterior y posterior
colNan_convert = ['tmed','prec','tmin','tmax']
for col in colNan_convert:
    madrid_clima3[col] = madrid_clima3[col].interpolate()

In [98]:
# Cambio el tipo de la columna fecha a tipo datetime
madrid_clima3["fecha"]=pd.to_datetime(madrid_clima3["fecha"],format="%Y-%m-%d")

In [205]:
TODAVIA NO HE DECIDICO SI ME INTERESA TENER FECHA COMO INDICE
#Establezco la fecha como índice
#madrid_clima.set_index('fecha',  inplace=True)

In [100]:
madrid_clima3.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 43830 entries, 0 to 1459
Data columns (total 20 columns):
 #   Column       Non-Null Count  Dtype         
---  ------       --------------  -----         
 0   fecha        43830 non-null  datetime64[ns]
 1   indicativo   43830 non-null  object        
 2   nombre       43830 non-null  object        
 3   provincia    43830 non-null  object        
 4   altitud      43830 non-null  int64         
 5   tmed         43830 non-null  float32       
 6   prec         43830 non-null  float32       
 7   tmin         43830 non-null  float32       
 8   horatmin     43807 non-null  object        
 9   tmax         43830 non-null  float32       
 10  horatmax     43818 non-null  object        
 11  dir          43258 non-null  float64       
 12  velmedia     43303 non-null  float32       
 13  racha        43168 non-null  float32       
 14  horaracha    43127 non-null  object        
 15  sol          37912 non-null  float32       
 16  presM

El tener un gran número de datos (datos diarios de 40 años) va a hacer ilegibles los gráficos de tendencia diarios por lo que lo pasaremos a una frecuencia mensual manteniendo los registros por estación. Separaremos el dataframe por estaciones, cambio las fechas a frecuencia  mensual y vuelvo a unirlo en un único dataframe

In [24]:
aeropuerto = madrid_clima[madrid_clima['indicativo']== '3129' ]
retiro = madrid_clima[madrid_clima['indicativo']== '3195' ]
cuatrovientos = madrid_clima[madrid_clima['indicativo']== '3196' ]
cuatrovientos.head(4)

fecha indicativo                  nombre provincia  altitud  tmed  \
2  1980-01-01       3196  MADRID, CUATRO VIENTOS    MADRID      690  11.8   
6  1980-01-02       3196  MADRID, CUATRO VIENTOS    MADRID      690   8.5   
10 1980-01-03       3196  MADRID, CUATRO VIENTOS    MADRID      690   5.5   
14 1980-01-04       3196  MADRID, CUATRO VIENTOS    MADRID      690   5.4   

    prec  tmin horatmin  tmax horatmax   dir  velmedia  racha horaracha  sol  \
2    0.0   8.4    04:40  15.2    13:20  27.0       5.3   11.1     16:20  7.7   
6    0.0   4.0    23:59  13.0    14:40  29.0       1.1   11.1     04:00  6.6   
10   0.0  -0.6    08:15  11.6    14:20  23.0       0.6    3.1     12:35  7.5   
14   0.0  -0.6    08:10  11.4    14:00  25.0       0.3    3.9     21:43  8.2   

       presMax horaPresMax     presMin horaPresMin  
2   939.700012          24  937.599976          06  
6   944.700012          24  939.700012          00  
10  945.900024          11  943.700012          24  
14  943.700012          00  940.500000          15

Las columnas en las que me interesa el valor medio cuando agrupe por meses son: tmed, prec, tmin, tmax, velmedia,sol
La columna en las que me interesa la suma de los registos durante el mes es: prec
Añadire nuevas columnas llamadas tmaxmes y rachames que serán los registros maximos de tmax y racha recogidos durante el mes.
Asimismo añadire otra columna llamada tminmes que sera el registro mínimo de temperatura registrado durante ese mes.


Elimino las columnas cuyos datos no voy a analizar (horatmin, horatmax, dir, horaracha, horaPresMax, horaPresMin)


Al tener un gran número de datos (datos diarios de 40 años) va a hacer ilegibles los gráficos de tendencia diarios por lo que lo pasaremos a una frecuencia mensual.

Las columnas en las que me interesa el valor medio cuando agrupe por meses son: tmed, tmin, tmax, velmedia,sol La columna en las que me interesa la suma de los registos durante el mes es: prec 

Añadire nuevas columnas llamadas tmaxmes y rachames que serán los registros maximos de tmax y racha recogidos durante el mes. Asimismo añadire otra columna llamada tminmes que sera el registro mínimo de temperatura registrado durante ese mes.



In [25]:
#Agrupo y me quedo solo con las columnas que dan información que posteriormente graficaré.
madrid_clima_1 = (madrid_clima[['fecha','nombre','tmed','prec','tmin','tmax','velmedia','racha','sol']]).groupby(['nombre','fecha'])['tmed','prec','tmin','tmax','velmedia','racha','sol'].sum()
madrid_clima_1

tmed  prec  tmin  tmax  velmedia  racha  sol
nombre            fecha                                                   
MADRID AEROPUERTO 1980-01-01  13.6   0.0  11.0  16.1       6.9   13.3  7.9
                  1980-01-02   7.7   0.0   3.4  12.0       1.9    8.9  3.8
                  1980-01-03   4.2   0.0  -2.8  11.3       1.1    3.3  6.6
                  1980-01-04   4.0   0.0  -2.6  10.6       0.0    0.0  6.1
                  1980-01-05   4.2   0.0  -2.2  10.6       0.0    2.2  4.9
...                            ...   ...   ...   ...       ...    ...  ...
MADRID, RETIRO    2019-12-27   8.9   0.0   4.3  13.5       0.8    7.5  0.0
                  2019-12-28   8.2   0.0   2.5  13.9       0.6    3.6  0.0
                  2019-12-29   7.8   0.0   2.5  13.2       0.8    3.9  0.0
                  2019-12-30   7.0   0.0   2.5  11.5       0.8    0.0  0.0
                  2019-12-31   6.8   0.0   2.3  11.4       0.3    3.1  0.0

[43830 rows x 7 columns]

In [26]:
#madrid_clima.rename(columns={'nombre':'estacion', 'sol':'horasol'}, inplace=True)
madrid_clima_1 = madrid_clima[['fecha','nombre','tmed','prec','tmin','tmax','velmedia','racha','sol']]
#madrid_clima_1
#madrid_tpp = madrid_clima_1.set_index('estación',inplace=True)
madrid_tpp= madrid_clima_1.groupby(['nombre','fecha'])['tmed','prec','tmin','tmax','velmedia','racha','sol'].sum()
madrid_tpp
#madrid_mensual_tpp= madrid_tpp.resample('M',level=0).mean().reset_index
#madrid_mensual_tpp

tmed  prec  tmin  tmax  velmedia  racha  sol
nombre            fecha                                                   
MADRID AEROPUERTO 1980-01-01  13.6   0.0  11.0  16.1       6.9   13.3  7.9
                  1980-01-02   7.7   0.0   3.4  12.0       1.9    8.9  3.8
                  1980-01-03   4.2   0.0  -2.8  11.3       1.1    3.3  6.6
                  1980-01-04   4.0   0.0  -2.6  10.6       0.0    0.0  6.1
                  1980-01-05   4.2   0.0  -2.2  10.6       0.0    2.2  4.9
...                            ...   ...   ...   ...       ...    ...  ...
MADRID, RETIRO    2019-12-27   8.9   0.0   4.3  13.5       0.8    7.5  0.0
                  2019-12-28   8.2   0.0   2.5  13.9       0.6    3.6  0.0
                  2019-12-29   7.8   0.0   2.5  13.2       0.8    3.9  0.0
                  2019-12-30   7.0   0.0   2.5  11.5       0.8    0.0  0.0
                  2019-12-31   6.8   0.0   2.3  11.4       0.3    3.1  0.0

[43830 rows x 7 columns]

In [27]:
madrid_tpp.sort_index(inplace=True)
madrid_tpp

tmed  prec  tmin  tmax  velmedia  racha  sol
nombre            fecha                                                   
MADRID AEROPUERTO 1980-01-01  13.6   0.0  11.0  16.1       6.9   13.3  7.9
                  1980-01-02   7.7   0.0   3.4  12.0       1.9    8.9  3.8
                  1980-01-03   4.2   0.0  -2.8  11.3       1.1    3.3  6.6
                  1980-01-04   4.0   0.0  -2.6  10.6       0.0    0.0  6.1
                  1980-01-05   4.2   0.0  -2.2  10.6       0.0    2.2  4.9
...                            ...   ...   ...   ...       ...    ...  ...
MADRID, RETIRO    2019-12-27   8.9   0.0   4.3  13.5       0.8    7.5  0.0
                  2019-12-28   8.2   0.0   2.5  13.9       0.6    3.6  0.0
                  2019-12-29   7.8   0.0   2.5  13.2       0.8    3.9  0.0
                  2019-12-30   7.0   0.0   2.5  11.5       0.8    0.0  0.0
                  2019-12-31   6.8   0.0   2.3  11.4       0.3    3.1  0.0

[43830 rows x 7 columns]

In [ ]:
madrid_tpp.reset_index('nombre').groupby('nombre', group_keys=True).resample('M').mean()

In [28]:
madrid_mes_mean = madrid_tpp.reset_index('nombre').groupby('nombre', group_keys=True).resample('M').mean()
madrid_mes_sum = madrid_tpp.reset_index('nombre').groupby('nombre', group_keys=True).resample('M').sum()
madrid_mes_max = madrid_tpp.reset_index('nombre').groupby('nombre', group_keys=True).resample('M').max()
madrid_mes_min = madrid_tpp.reset_index('nombre').groupby('nombre', group_keys=True).resample('M').min()
madrid_mes_sum

tmed       prec        tmin         tmax  \
nombre            fecha                                                        
MADRID AEROPUERTO 1980-01-31  170.699997  11.600000   36.200001   305.299988   
                  1980-02-29  220.199997  49.200001   61.500000   378.899994   
                  1980-03-31  303.799988  43.333332  144.500000   462.700012   
                  1980-04-30  337.600006  35.700001  135.600006   539.400024   
                  1980-05-31  430.200012  78.666664  251.399994   608.700012   
...                                  ...        ...         ...          ...   
MADRID, RETIRO    2019-08-31  808.000000  58.400002  610.599976  1005.799988   
                  2019-09-30  635.700012  33.099998  470.299988   800.500000   
                  2019-10-31  501.700012  42.599998  361.000000   642.299988   
                  2019-11-30  288.200012  44.000000  199.199997   377.399994   
                  2019-12-31  272.799988  90.099998  173.899994   371.500000   

                                velmedia       racha         sol  
nombre            fecha                                           
MADRID AEROPUERTO 1980-01-31   90.699997  211.199997  112.300003  
                  1980-02-29   45.299999  161.699997  124.300003  
                  1980-03-31  131.300003  296.700012  199.600006  
                  1980-04-30  116.000000  303.000000  219.199997  
                  1980-05-31  107.300003  257.899994  239.000000  
...                                  ...         ...         ...  
MADRID, RETIRO    2019-08-31   60.700001  293.299988    0.000000  
                  2019-09-30   63.099998  290.299988    0.000000  
                  2019-10-31   47.700001  236.500000    0.000000  
                  2019-11-30   79.199997  330.100006    0.000000  
                  2019-12-31   63.400002  268.100006    0.000000  

[1440 rows x 7 columns]

In [29]:
madrid_mesmean =madrid_mes_mean[['tmed','tmin','tmax','velmedia','sol']].rename(columns={'tmin':'tmin_media','tmax':'tmax_media','sol':'solmedia'})
madrid_mesmean

tmed  tmin_media  tmax_media  velmedia  \
nombre            fecha                                                     
MADRID AEROPUERTO 1980-01-31   5.506452    1.167742    9.848387  2.925807   
                  1980-02-29   7.593103    2.120690   13.065517  1.562069   
                  1980-03-31   9.800000    4.661290   14.925806  4.235484   
                  1980-04-30  11.253333    4.520000   17.980000  3.866667   
                  1980-05-31  13.877419    8.109677   19.635485  3.461290   
...                                 ...         ...         ...       ...   
MADRID, RETIRO    2019-08-31  26.064516   19.696774   32.445160  1.958065   
                  2019-09-30  21.190001   15.676666   26.683334  2.103333   
                  2019-10-31  16.183870   11.645162   20.719355  1.538710   
                  2019-11-30   9.606667    6.640000   12.580000  2.640000   
                  2019-12-31   8.800000    5.609677   11.983871  2.045161   

                              solmedia  
nombre            fecha                 
MADRID AEROPUERTO 1980-01-31  3.622581  
                  1980-02-29  4.286207  
                  1980-03-31  6.438710  
                  1980-04-30  7.306667  
                  1980-05-31  7.709677  
...                                ...  
MADRID, RETIRO    2019-08-31  0.000000  
                  2019-09-30  0.000000  
                  2019-10-31  0.000000  
                  2019-11-30  0.000000  
                  2019-12-31  0.000000  

[1440 rows x 5 columns]

In [30]:
madrid_messum = madrid_mes_sum[['prec']].rename(columns={'prec':'precmes'})
#madrid_messum = planets.rename(columns={'periodo_orbital':'periodo orbital','método':'método descubrimiento'})
madrid_messum 

precmes
nombre            fecha                
MADRID AEROPUERTO 1980-01-31  11.600000
                  1980-02-29  49.200001
                  1980-03-31  43.333332
                  1980-04-30  35.700001
                  1980-05-31  78.666664
...                                 ...
MADRID, RETIRO    2019-08-31  58.400002
                  2019-09-30  33.099998
                  2019-10-31  42.599998
                  2019-11-30  44.000000
                  2019-12-31  90.099998

[1440 rows x 1 columns]

In [31]:
madrid_mesmax = madrid_mes_max[['tmax','prec','racha']]
madrid_mesmax 

tmax       prec      racha
nombre            fecha                                      
MADRID AEROPUERTO 1980-01-31  19.799999   7.900000  16.700001
                  1980-02-29  18.000000   9.000000  13.900000
                  1980-03-31  24.900000  21.299999  17.799999
                  1980-04-30  26.600000  11.300000  15.600000
                  1980-05-31  25.600000  26.600000  14.400000
...                                 ...        ...        ...
MADRID, RETIRO    2019-08-31  37.200001  26.900000  15.300000
                  2019-09-30  32.700001  13.700000  15.800000
                  2019-10-31  28.000000  18.400000  17.200001
                  2019-11-30  19.299999   8.800000  17.799999
                  2019-12-31  16.000000  38.400002  20.600000

[1440 rows x 3 columns]

In [32]:
madrid_mesmin = madrid_mes_min[['tmin']] 
madrid_mesmin

tmin
nombre            fecha           
MADRID AEROPUERTO 1980-01-31  -5.4
                  1980-02-29  -3.0
                  1980-03-31  -3.4
                  1980-04-30  -0.4
                  1980-05-31   4.0
...                            ...
MADRID, RETIRO    2019-08-31  15.0
                  2019-09-30  12.0
                  2019-10-31   6.3
                  2019-11-30   0.7
                  2019-12-31   2.3

[1440 rows x 1 columns]

Concateno horizontalmente los tres data de cada estación, redondeo los valores a 1 decimal y añado los indices como columnas en el nuevo dataframe

In [81]:
#Concateno horizontalmente los tres data de cada estación, redondeo los valores a 1 decimal y añado los indices como columnas en el nuevo dataframe 
madrid_mensual3 = pd.concat([madrid_mesmean, madrid_messum, madrid_mesmax ,madrid_mesmin], axis=1).reindex(madrid_mesmean.index)

In [82]:
colums_round = ('tmed',	'tmin_media',	'tmax_media',	'velmedia',	'solmedia',	'precmes',	'tmax',	'prec',	'racha')
for elem in colums_round:
    madrid_mensual3[elem] = madrid_mensual3[elem].apply(lambda x: round(x, 1))

In [60]:
for elem in colums_round:
    madrid_trabajo[elem] = madrid_trabajo[elem].apply(lambda x: round(x, 1))

In [85]:
madrid_mensual3.reset_index(inplace=True)

In [86]:
madrid_mensual3

nombre      fecha  tmed  tmin_media  tmax_media  velmedia  \
0     MADRID AEROPUERTO 1980-01-31   5.5         1.2         9.8       2.9   
1     MADRID AEROPUERTO 1980-02-29   7.6         2.1        13.1       1.6   
2     MADRID AEROPUERTO 1980-03-31   9.8         4.7        14.9       4.2   
3     MADRID AEROPUERTO 1980-04-30  11.3         4.5        18.0       3.9   
4     MADRID AEROPUERTO 1980-05-31  13.9         8.1        19.6       3.5   
...                 ...        ...   ...         ...         ...       ...   
1435     MADRID, RETIRO 2019-08-31  26.1        19.7        32.4       2.0   
1436     MADRID, RETIRO 2019-09-30  21.2        15.7        26.7       2.1   
1437     MADRID, RETIRO 2019-10-31  16.2        11.6        20.7       1.5   
1438     MADRID, RETIRO 2019-11-30   9.6         6.6        12.6       2.6   
1439     MADRID, RETIRO 2019-12-31   8.8         5.6        12.0       2.0   

      solmedia  precmes  tmax  prec  racha  tmin  
0          3.6     11.6  19.8   7.9   16.7  -5.4  
1          4.3     49.2  18.0   9.0   13.9  -3.0  
2          6.4     43.3  24.9  21.3   17.8  -3.4  
3          7.3     35.7  26.6  11.3   15.6  -0.4  
4          7.7     78.7  25.6  26.6   14.4   4.0  
...        ...      ...   ...   ...    ...   ...  
1435       0.0     58.4  37.2  26.9   15.3  15.0  
1436       0.0     33.1  32.7  13.7   15.8  12.0  
1437       0.0     42.6  28.0  18.4   17.2   6.3  
1438       0.0     44.0  19.3   8.8   17.8   0.7  
1439       0.0     90.1  16.0  38.4   20.6   2.3  

[1440 rows x 12 columns]

In [91]:
madrid_mensual3[madrid_mensual3['nombre']=='MADRID AEROPUERTO']

nombre      fecha  tmed  tmin_media  tmax_media  velmedia  \
0    MADRID AEROPUERTO 1980-01-31   5.5         1.2         9.8       2.9   
1    MADRID AEROPUERTO 1980-02-29   7.6         2.1        13.1       1.6   
2    MADRID AEROPUERTO 1980-03-31   9.8         4.7        14.9       4.2   
3    MADRID AEROPUERTO 1980-04-30  11.3         4.5        18.0       3.9   
4    MADRID AEROPUERTO 1980-05-31  13.9         8.1        19.6       3.5   
..                 ...        ...   ...         ...         ...       ...   
475  MADRID AEROPUERTO 2019-08-31  25.7        17.8        33.7       3.3   
476  MADRID AEROPUERTO 2019-09-30  21.0        14.1        27.9       3.0   
477  MADRID AEROPUERTO 2019-10-31  16.1         9.6        22.6       2.6   
478  MADRID AEROPUERTO 2019-11-30  10.2         6.6        13.8       5.0   
479  MADRID AEROPUERTO 2019-12-31   8.4         3.9        13.0       3.2   

     solmedia  precmes  tmax  prec  racha  tmin  
0         3.6     11.6  19.8   7.9   16.7  -5.4  
1         4.3     49.2  18.0   9.0   13.9  -3.0  
2         6.4     43.3  24.9  21.3   17.8  -3.4  
3         7.3     35.7  26.6  11.3   15.6  -0.4  
4         7.7     78.7  25.6  26.6   14.4   4.0  
..        ...      ...   ...   ...    ...   ...  
475      10.4     44.4  37.6  33.3   20.6  14.0  
476       8.5     37.9  34.4  14.1   20.0   9.7  
477       7.2     33.1  30.1  12.1   23.3   3.6  
478       3.9     32.1  21.6   7.3   21.1  -2.4  
479       4.0     58.0  17.6  26.0   24.2  -2.3  

[480 rows x 12 columns]

In [90]:
madrid_mensual3.isnull().sum()

nombre        0
fecha         0
tmed          0
tmin_media    0
tmax_media    0
velmedia      0
solmedia      0
precmes       0
tmax          0
prec          0
racha         0
tmin          0
dtype: int64

In [ ]:
tmed	tmin_media	tmax_media	velmedia	solmedia	precmes	tmax	prec	racha	tmin

## 4. Análisis de tendencias en los datos (EDA)
________________

In [87]:
#Lista de las columnas de datos para graficarlas y contrastar la tendencia entre las distintas estaciones
cols = ['tmed', 'precmes',
       'tmin_media', 'tmax_media', 'velmedia',
       'solmedia']

Tendencias de temperatura y precipitación.

In [88]:
for elem in cols:
    vtb.grafI_est("fecha" , elem , "nombre", madrid_mensual3, False )

In [102]:
madrid3_1 = madrid_clima3.copy()
madrid3_1

fecha indicativo                  nombre provincia  altitud  tmed  \
0    1980-01-01       3129       MADRID AEROPUERTO    MADRID      609  13.6   
2    1980-01-01       3196  MADRID, CUATRO VIENTOS    MADRID      690  11.8   
3    1980-01-01       3195          MADRID, RETIRO    MADRID      667  12.5   
4    1980-01-02       3129       MADRID AEROPUERTO    MADRID      609   7.7   
6    1980-01-02       3196  MADRID, CUATRO VIENTOS    MADRID      690   8.5   
...         ...        ...                     ...       ...      ...   ...   
1454 2019-12-30       3196  MADRID, CUATRO VIENTOS    MADRID      690   7.4   
1455 2019-12-30       3195          MADRID, RETIRO    MADRID      667   7.0   
1456 2019-12-31       3129       MADRID AEROPUERTO    MADRID      609   5.8   
1458 2019-12-31       3196  MADRID, CUATRO VIENTOS    MADRID      690   7.3   
1459 2019-12-31       3195          MADRID, RETIRO    MADRID      667   6.8   

      prec  tmin horatmin  tmax horatmax   dir  velmedia  racha horaracha  \
0      0.0  11.0   Varias  16.1    14:30  29.0       6.9   13.3    Varias   
2      0.0   8.4    04:40  15.2    13:20  27.0       5.3   11.1     16:20   
3      0.0   9.7    06:15  15.3    13:30  29.0       5.3   11.1     11:20   
4      0.0   3.4    06:00  12.0    14:00  27.0       1.9    8.9     00:00   
6      0.0   4.0    23:59  13.0    14:40  29.0       1.1   11.1     04:00   
...    ...   ...      ...   ...      ...   ...       ...    ...       ...   
1454   0.0   1.5    05:10  13.2    14:10  14.0       0.6    4.2     11:30   
1455   0.0   2.5    05:30  11.5   Varias   NaN       0.8    NaN       NaN   
1456   0.0  -1.4    07:05  12.9    14:28  17.0       0.6    4.7     14:48   
1458   0.0   1.5    07:30  13.1   Varias  35.0       0.8    3.1     02:01   
1459   0.0   2.3    07:30  11.4    16:00   7.0       0.3    3.1     09:30   

      sol     presMax horaPresMax     presMin horaPresMin  
0     7.9  950.900024          23  948.700012      Varias  
2     7.7  939.700012          24  937.599976          06  
3     8.2         NaN         NaN         NaN         NaN  
4     3.8  957.599976          23  950.900024      Varias  
6     6.6  944.700012          24  939.700012          00  
...   ...         ...         ...         ...         ...  
1454  8.9  951.900024          11  949.500000      Varias  
1455  NaN  954.000000          11  950.900024          00  
1456  8.3  964.400024      Varias  961.700012          14  
1458  8.7  952.000000      Varias  949.700012      Varias  
1459  NaN  954.200012          10  951.799988      Varias  

[43830 rows x 20 columns]

In [1]:
madrid_total = madrid3_1.groupby('fecha')['tmed','prec','tmin','tmax','velmedia','racha','sol'].mean()
madrid_total.

NameError: name 'madrid3_1' is not defined

In [75]:
col_dat1 = ["total_cases", "total_deaths", "total_cases_per_million"]
col_dat2 = [ "rank_TC", "rank_TD", "rank_TCxM", "rank_TDxM"]
for elem in paises_grupoD:
    df = covid_grupoD[covid_grupoD.iso_code==elem]
    vtb.grafI_pais(elem, col_dat1, paises_nombres[elem], df, True,alarm_dates)
    vtb.grafI_pais(elem, col_dat2, paises_nombres[elem], df, False, alarm_dates) 
    vtb.graf_daily ('New daily cases',"new_cases", 'Daily deaths', "new_deaths", elem ,df, paises_nombres[elem],alarm_dates )
    vtb.graf_dailyd ('Daily deaths', "new_deaths", elem ,df, paises_nombres[elem])

NameError: name 'paises_grupoD' is not defined

In [76]:
covid_general.index

NameError: name 'covid_general' is not defined

In [77]:
# Revisión de los datos globales para ver si hay información totalizada (continent or country tienen valor  Null) y eliminarla posteriormente para el análisis
covid_general[(covid_general.continent.isnull()) | (covid_general.iso_code.isnull())]

NameError: name 'covid_general' is not defined

In [78]:
# DF Mundial sin los datos globales (continent is Nan) o de países Nan
world_df =  covid_general[~((covid_general.continent.isnull()) | (covid_general.iso_code.isnull()))]

NameError: name 'covid_general' is not defined

In [79]:
world_df.shape

NameError: name 'world_df' is not defined

In [80]:
world_df.info()

NameError: name 'world_df' is not defined

In [81]:
world_df["date"]=pd.to_datetime(world_df["date"],format="%Y-%m-%d")

NameError: name 'world_df' is not defined

______________
## RANKING CON RESPECTO AL TOTAL
What position do your countries occupe respect to the number of total infected, total deaths and total recoveries?
______________

In [82]:
#Ranking diario por países considerando el numero de casos totales (i.e. Total Infected)
world_df["rank_TC"] = world_df.groupby(["date"])["total_cases"].rank(method = "dense", ascending = False)

NameError: name 'world_df' is not defined

In [83]:
#Ranking diario por países considerando el numero de muertes (i.e. Total Deaths)
world_df["rank_TD"] = world_df.groupby(["date"])["total_deaths"].rank(method = "dense", ascending = False)

NameError: name 'world_df' is not defined

In [84]:
#Ranking diario por países considerando el numero de casos totales por millon de habitantes (i.e. total_cases_per_million)
world_df["rank_TCxM"] = world_df.groupby(["date"])["total_cases_per_million"].rank(method = "dense", ascending = False)

NameError: name 'world_df' is not defined

In [85]:
#Ranking diario por países considerando el numero de muertes totales por millon de habitantes (i.e. total_deaths_per_million)
world_df["rank_TDxM"] = world_df.groupby(["date"])["total_deaths_per_million"].rank(method = "dense", ascending = False)

NameError: name 'world_df' is not defined

In [86]:
#EJEMPLO DEL RANKING DE CASOS TOTALES PARA EL DIA 28 DE MARZO
#world_df.iloc[world_df[world_df.date=="2020-03-28"]["rank_TC"].sort_values(ascending=True).index]

In [87]:
#EJEMPLO DEL RANKING DE CASOS TOTALES PARA ESPAñA
world_df[world_df.iso_code=="ESP"].tail(5)

NameError: name 'world_df' is not defined

In [88]:
#EJEMPLO DEL RANKING para PORTUGAL  
world_df[world_df.iso_code=="PRT"].tail(5)

NameError: name 'world_df' is not defined

In [89]:
# EJEMPLO DEL RANKING para TURQUIA
world_df[world_df.iso_code=="TUR"].tail(5)

NameError: name 'world_df' is not defined

In [90]:
# EJEMPLO DEL RANKING para VENEZUELA
world_df[world_df.iso_code=="VEN"].tail(5)

NameError: name 'world_df' is not defined

In [91]:
# EJEMPLO DEL RANKING para GRAN BRETAÑA
world_df[world_df.iso_code=="GBR"].tail(5)

NameError: name 'world_df' is not defined

In [92]:
cols_rank= ["rank_TC", "rank_TD", "rank_TCxM", "rank_TDxM" ]

## ANÁLISIS CONJUNTO PAISES DEL GRUPO D
### PORTUGAL, VENEZUELA, TURQUÍA, GRAN BRETAÑA y ESPAÑA

Filtro (en un dataframe) de los datos correspondientes a los países del grupo D

In [93]:
paises_grupoD=["PRT", "VEN", "TUR", "GBR", "ESP"]
paises_nombres={"PRT":"Portugal", "VEN":"Venezuela", "TUR":"Turquía", "GBR":"Gran Bretaña", "ESP":"España"}
covid_grupoD=world_df[world_df.iso_code.isin(paises_grupoD)]
covid_grupoD


NameError: name 'world_df' is not defined

In [94]:
covid_grupoD.info()

NameError: name 'covid_grupoD' is not defined

In [95]:
covid_grupoD["date"].min()

NameError: name 'covid_grupoD' is not defined

In [96]:
#Selección de los registros a partir de la fecha en que comenzaron a reportar  datos de COVID
covid_grupoD[covid_grupoD["date"] == covid_grupoD["date"].min()]

NameError: name 'covid_grupoD' is not defined

In [97]:
covid_grupoD.new_cases.value_counts(dropna = False).sum()

NameError: name 'covid_grupoD' is not defined

In [98]:
# Crear un DF con las filas de datos que No contienen informacion del COVID
nocovid_grupoD = covid_grupoD[(covid_grupoD["new_cases"]==0) & (covid_grupoD["total_cases"] ==0)]

NameError: name 'covid_grupoD' is not defined

In [99]:
nocovid_grupoD.shape

NameError: name 'nocovid_grupoD' is not defined

In [100]:
covid_grupoD.isnull().sum()

NameError: name 'covid_grupoD' is not defined

In [101]:
# Eliminar del DF las filas que NO contienen info del COVID
covid_grupoD=covid_grupoD.drop(covid_grupoD[(covid_grupoD["new_cases"]==0) & (covid_grupoD["total_cases"] ==0)].index)

NameError: name 'covid_grupoD' is not defined

In [102]:
covid_grupoD.shape

NameError: name 'covid_grupoD' is not defined

In [103]:
# Eliminar la columna handwashing_facilities pues no contiend informacion
covid_grupoD.drop(columns=["handwashing_facilities"], axis = 1, inplace=True)

NameError: name 'covid_grupoD' is not defined

In [104]:
covid_grupoD.groupby("iso_code")["date"].min()

NameError: name 'covid_grupoD' is not defined

In [105]:
covid_grupoD.reset_index(inplace=True)

NameError: name 'covid_grupoD' is not defined

In [106]:
covid_grupoD.info()

NameError: name 'covid_grupoD' is not defined

In [107]:
# Creacion de un DF con las fechas del estado de alarma para cada país
alarm_dicc = {'iso_code': ["ESP", "PRT", "VEN", "GBR", "TUR"],
        'alarm_init': ['2020-03-14', '2020-03-13', '2020-03-13', '2020-03-23',np.NaN],
        'alarm_end': ['2020-06-21', '2020-06-01', '2020-07-11', '2020-07-04', np.NaN]
       }
alarm_dates = pd.DataFrame(alarm_dicc, columns = ['iso_code', 'alarm_init', 'alarm_end'])

alarm_dates

iso_code  alarm_init   alarm_end
0      ESP  2020-03-14  2020-06-21
1      PRT  2020-03-13  2020-06-01
2      VEN  2020-03-13  2020-07-11
3      GBR  2020-03-23  2020-07-04
4      TUR         NaN         NaN

# Show different tendencies for each column in your dataset

In [108]:
#Lista de las columnas de datos para graficarlas y contrastar la tendencia entre los países del grupo D
cols = list(covid_grupoD.columns.values[[5,6,7,8,9,10,11,12,13,14,15,16,17]])
cols

NameError: name 'covid_grupoD' is not defined

In [109]:
# Función para mostrar el valor de una columna de datos en un gráfico de líneas (1 por país)
def graf_grupo (x_val, y_val, hue_val, df, inv_y): 
    fig= plt.figure(figsize =(30,10))
    g = sns.relplot(x =x_val, y = y_val, hue = hue_val, kind = "line", data = df, palette = "Paired")
    g._legend.texts[0].set_text("")
    g._legend.set_title("Country")
    plt.xticks(rotation = "vertical")
    tit = y_val.upper() + " by Date and Country - Group D"
    plt.title(tit)
    plt.savefig("../resources/" + y_val+ "_x_C_D" +  ".png",bbox_inches='tight')
    if inv_y:
        plt.gca().invert_yaxis()
    plt.show()

In [110]:
for elem in cols:
    graf_grupo("date" , elem , "iso_code", covid_grupoD, False )

NameError: name 'covid_grupoD' is not defined

In [111]:
#Gráficos de cada variable separados por país
for elem in cols:
    sns.relplot(x="date", y=elem, col="iso_code", kind="line", data=covid_grupoD)

NameError: name 'covid_grupoD' is not defined

______________
## ANÁLISIS DETALLADO POR PAÍS
Show different tendencies for each column in your dataset. Show, vertically, the start date and end date of the alarm state in each plot
______________

### MUERTES TOTALES Y CASOS TOTALES DE CADA PAÍS (CONTRASTADOS CON FECHAS DE ESTADO DE ALARMA)

In [112]:
def graf_pais (ccode,col_dat, cname,cdf, y_maximo):
    if y_maximo:
        eje_y = cdf.total_cases.max()
        tit = cname.upper() + " COVID19 - TOTALES"
        filen= "Tot_"
    else:
        eje_y = cdf.rank_TCxM.max()
        tit = cname.upper() + " COVID19 - RANKING"
        filen= "Rank_"
    fg = cdf.plot(x="date", y= col_dat, kind="line", figsize=(12,8),title= tit)
    fg.legend(bbox_to_anchor=(1.2, 0.5))
    plt.vlines(alarm_dates.alarm_init[alarm_dates.iso_code == ccode].to_frame().iloc[0,0], ymin=0, ymax=eje_y, color = "r", linestyles =":")
    plt.vlines(alarm_dates.alarm_end[alarm_dates.iso_code == ccode].to_frame().iloc[0,0], ymin=0, ymax=eje_y, color = "r", linestyles =":")
    plt.savefig("../resources/"+ filen + elem+".png",bbox_inches='tight')
    plt.show()

In [113]:
sns.set()
col_dat1 = ["total_deaths", "total_cases", "total_cases_per_million"]
col_dat2 = [ "rank_TD", "rank_TC", "rank_TCxM", "rank_TDxM"]
for elem in paises_grupoD:
    df = covid_grupoD[covid_grupoD.iso_code==elem]
    graf_pais(elem, col_dat1, paises_nombres[elem], df, True)
    graf_pais(elem, col_dat2, paises_nombres[elem], df, False)
  


NameError: name 'covid_grupoD' is not defined

In [114]:
# Como han variado los paises en el rank_TC
lp = sns.lineplot('date','rank_TC', ci=None, hue='location', palette="bright",data=covid_grupoD) # dibujo grafico
lp.legend(loc='upper right', bbox_to_anchor=(1.4,1)) # cuadro de la leyenda 

NameError: name 'covid_grupoD' is not defined

In [115]:
# Como han variado los paises en el rank_TD
lp = sns.lineplot('date','rank_TD', ci=None, hue='location', palette="bright",data=covid_grupoD) # dibujo grafico
lp.legend(loc='upper right', bbox_to_anchor=(1.4,1)) # cuadro de la leyenda 

NameError: name 'covid_grupoD' is not defined

In [116]:
# Como han variado los paises en el rank_TCxM
lp = sns.lineplot('date','rank_TCxM', ci=None, hue='location', palette="bright",data=covid_grupoD) # dibujo grafico
lp.legend(loc='upper right', bbox_to_anchor=(1.4,1)) # cuadro de la leyenda 

NameError: name 'covid_grupoD' is not defined

In [117]:
# Como han variado los paises en el rank_TDxM
lp = sns.lineplot('date','rank_TDxM', ci=None, hue='location', palette="bright",data=covid_grupoD) # dibujo grafico
lp.legend(loc='upper right', bbox_to_anchor=(1.4,1)) # cuadro de la leyenda 

NameError: name 'covid_grupoD' is not defined

In [118]:
# La idea es obtener un cuadro con el ranking que ocupa cada pais el ultimo de los días con datos, pero no es correcto porque nos da el valor máximo que a lo largo del tiempo a tenido cada país en los rankings.
covid_grupoD.groupby(['iso_code'])['date','rank_TC','rank_TD', 'rank_TDxM', 'rank_TCxM'].max()
#¿Teneis alguna idea de como se podría hacer?

NameError: name 'covid_grupoD' is not defined

In [119]:
covid_grupoD[covid_grupoD.iso_code =='ESP'].tail(3)

NameError: name 'covid_grupoD' is not defined

In [120]:
#Dataframe con información resumida de datos referenciales por país
datos_paises = covid_grupoD.iloc[:,[1,20,21,22,23,24,29,30,32,33]].drop_duplicates()

NameError: name 'covid_grupoD' is not defined

In [121]:
datos_paises.info()
# Nota: 
# Stringency Index: The Government Response Stringency Index is a composite measure based on nine response indicators including
# school closures, workplace closures, and travel bans, rescaled to a value from 0 to 100 (100 = strictest response).

NameError: name 'datos_paises' is not defined

In [122]:
datos_paises

NameError: name 'datos_paises' is not defined

_________
## CORRELACIÓN
______________

In [123]:
covid_grupoD.info()

NameError: name 'covid_grupoD' is not defined

In [124]:
#Matriz de correlación entre variables
grupoD_corr = covid_grupoD.iloc[:,np.r_[1,5,7,9,11,22:36]]     #Un subconjunto de las variables a contrastar

NameError: name 'covid_grupoD' is not defined

In [125]:
grupoD_corr.info()

NameError: name 'grupoD_corr' is not defined

In [126]:
grupoD_corr.shape

NameError: name 'grupoD_corr' is not defined

In [127]:
f, ax = plt.subplots(figsize=(10, 8)) 
corr = grupoD_corr.corr() 
sns.heatmap(corr, mask=np.zeros_like(corr, dtype=np.bool), cmap=sns.diverging_palette(220, 10, as_cmap=True), 
      square=True, ax=ax, annot=True) 

NameError: name 'grupoD_corr' is not defined

## Matriz de correlación Todos los países.

In [128]:
#Seleccion del DF mundial del subconjunto de las variables a correlacionar
world_df_corr = world_df.iloc[:,np.r_[0,4,6,8,10,21:33,34:36]]    

NameError: name 'world_df' is not defined

In [129]:
#Mapa de calor de la matriz de correlación 
f, ax = plt.subplots(figsize=(10, 8)) 
corr2 = world_df_corr.corr() 
sns.heatmap(corr, mask=np.zeros_like(corr, dtype=np.bool), cmap=sns.diverging_palette(220, 10, as_cmap=True), 
      square=True, ax=ax, annot=True) 

NameError: name 'world_df_corr' is not defined

### NOTA:Otras pruebas. No considerar este código

In [130]:
#Dejar únicamente las columnas que interesan para la evaluación del ranking
#world_df = world_df[["iso_code", "continent", "location", "date", "total_cases", "total_deaths", "total_cases_per_million", "stringency_index", "population", "population_density","hospital_beds_per_thousand", "life_expectancy" ]]

In [131]:
#Crea un DF por cada país (llamado covid_ESP, etc) a partir de los datos del dataframe covid_grupoD. 
#for elem in paises_grupoD:
#    df_name = "covid_" + elem
#    globals()[df_name] = covid_grupoD[covid_grupoD.iso_code==elem]

In [132]:
#for elem in cols_rank:
#    sns.relplot(x="date", y=elem, col="iso_code", kind="line", data=rank_grupoD)

In [133]:
# Pruebas de Jose Luis.
# Para graficar total deaths de todos los paises.
lp = sns.lineplot('date','total_deaths', ci=None, hue='iso_code', palette="bright",data=world_df)
lp.legend(loc='upper right', bbox_to_anchor=(1.4,3))

# Para las líneas verticales del estado de alarma de un pais
#xposition = [pd.to_datetime('2020-03-14'), pd.to_datetime('2020-06-16')]
#for xc in xposition:
#    lp.axvline(x=xc, color='red', linestyle='--')

NameError: name 'world_df' is not defined

In [134]:
# Otra forma de graficar las total deaths de todos los paises.
fig = px.line(world_df, x="date", y="total_deaths", color="continent", line_group="location", hover_name="location",
        line_shape="spline", render_mode="svg")
fig.show()

NameError: name 'world_df' is not defined

In [135]:
# Para graficar total cases de todos los paises.
lp = sns.lineplot('date','total_cases', ci=None, hue='iso_code', palette="bright",data=world_df)
lp.legend(loc='upper right', bbox_to_anchor=(1.4,3))

NameError: name 'world_df' is not defined

In [136]:
# Otra forma de graficar las total cases de todos los paises.
fig = px.line(world_df, x="date", y="total_cases", color="continent", line_group="location", hover_name="location",
        line_shape="spline", render_mode="svg")
fig.show()

NameError: name 'world_df' is not defined

In [137]:
# Para graficar total deaths per million de todos los paises.
lp = sns.lineplot('date','total_deaths_per_million', ci=None, hue='iso_code', palette="bright",data=world_df)
lp.legend(loc='upper right', bbox_to_anchor=(1.4,3))

NameError: name 'world_df' is not defined

In [138]:
# Otra forma de graficar las total deaths per million de todos los paises.
# Como efecto secundario en la gráfica se puede ver el ranking y ohh sorpresa el nº1 es... San Marino.
fig = px.line(world_df, x="date", y="total_deaths_per_million", color="continent", line_group="location", hover_name="location",
        line_shape="spline", render_mode="svg")
fig.show()

NameError: name 'world_df' is not defined

In [139]:
# Para poder establecer algún tipo de relación entre población de +65 y muertes por covid
# paises_grupoD=["PRT", "VEN", "TUR", "GBR", "ESP"]
# paises_nombres={"PRT":"Portugal", "VEN":"Venezuela", "TUR":"Turquía", "GBR":"Gran Bretaña", "ESP":"España"}
# covid_grupoD=world_df[world_df.iso_code.isin(paises_grupoD)]
covid_grupoD.groupby('iso_code')['total_deaths'].max()

NameError: name 'covid_grupoD' is not defined

In [140]:
# Añado la columna del indice de mortalidad al df
covid_grupoD['mortality_rate'] = round((covid_grupoD['total_deaths']/covid_grupoD['total_cases'])*100, 2)

NameError: name 'covid_grupoD' is not defined

In [141]:
covid_grupoD.tail()

NameError: name 'covid_grupoD' is not defined

In [142]:
# Como a variado el indice de mortalidad
lp = sns.lineplot('date','mortality_rate', ci=None, hue='location', palette="bright",data=covid_grupoD) # dibujo grafico
lp.legend(loc='upper right', bbox_to_anchor=(1.4,1)) # cuadro de la leyenda 

NameError: name 'covid_grupoD' is not defined

In [143]:
# Igual que la anterior pero con otro metodo 
fig = px.line(covid_grupoD, x="date", y="mortality_rate", color="location", line_group="location", hover_name="location",
        line_shape="spline", render_mode="svg")
fig.show()

NameError: name 'covid_grupoD' is not defined

In [144]:
world_df_map = (world_df.groupby('location')['total_cases'].max()).to_frame()
world_df_map['index'] = world_df_map.index
world_df_map.rename(columns={'index': 'country'}, inplace=True)
world_df_map

NameError: name 'world_df' is not defined

In [145]:
fig = px.choropleth(world_df_map, locations=world_df_map['country'],
                    color=world_df_map['total_cases'],locationmode='country names', 
                    hover_name=world_df_map['country'], 
                    color_continuous_scale=px.colors.sequential.Tealgrn,template='plotly_white', )
fig.update_layout(title='Total Cases In Each Country',)
fig.show()

NameError: name 'world_df_map' is not defined

In [146]:
world_df_map_td = (world_df.groupby('location')['total_deaths'].max()).to_frame()
world_df_map_td['index'] = world_df_map_td.index
world_df_map_td.rename(columns={'index': 'country'}, inplace=True)

NameError: name 'world_df' is not defined

In [147]:
fig = px.choropleth(world_df_map_td, locations=world_df_map_td['country'],
                    color=world_df_map_td['total_deaths'],locationmode='country names', 
                    hover_name=world_df_map_td['country'], 
                    color_continuous_scale=px.colors.sequential.Tealgrn,template='plotly_white', )
fig.update_layout(title='Total Deaths in each Country',)
fig.show()

NameError: name 'world_df_map_td' is not defined

In [148]:
# Obtengo dataframe de España desde el dia que se notifica el primer caso en España.
covid_grupoD_ESP = covid_grupoD[(covid_grupoD['iso_code'] == 'ESP') & (covid_grupoD['total_cases']>0)] 
covid_grupoD_ESP.head(3)

NameError: name 'covid_grupoD' is not defined

In [149]:
# Grafico de barras de España con los casos y las muertes diarias.
fig = go.Figure(data=[
    go.Bar(name='New daily cases', x=covid_grupoD_ESP['date'], y=covid_grupoD_ESP['new_cases']),
    go.Bar(name='Daily deaths', x=covid_grupoD_ESP['date'], y=covid_grupoD_ESP['new_deaths'])])
# Change the bar mode
fig.update_layout(barmode='overlay', title='Daily Case and Daily Death en España',
                 annotations=[dict(x='2020-03-15', y=1407, xref="x", yref="y", text="Lockdown started", showarrow=True, arrowhead=1, ax=-50, ay=-70)])
fig.show()

NameError: name 'covid_grupoD_ESP' is not defined

In [150]:
import plotly.express as px
import time
import plotly.graph_objs as go
import plotly.offline as py
import plotly.io as pio
py.offline.init_notebook_mode(connected=True)

In [151]:
# Obtengo dataframe de Gran Bretaña desde el dia que se notifica el primer caso.
covid_grupoD_GBR = covid_grupoD[(covid_grupoD['iso_code'] == 'GBR') & (covid_grupoD['total_cases']>0)] 
covid_grupoD_GBR.head()

NameError: name 'covid_grupoD' is not defined

Grafico de barras animado que muestra como aumenta diariamente el número de new_cases

In [152]:
covid_grupoD.info()

NameError: name 'covid_grupoD' is not defined

In [153]:
covid_grupoD_infected = covid_grupoD.iloc[:,[1,3,4,5,6,7,8,22,23]]
covid_grupoD_infected

NameError: name 'covid_grupoD' is not defined

In [154]:
covid_grupoD_infected["new_date"] = covid_grupoD_infected.date.apply(lambda x: x.strftime("%d %b, %Y"))
covid_grupoD_infected.fillna(0, inplace=True)

NameError: name 'covid_grupoD_infected' is not defined

In [155]:
total_s = covid_grupoD_infected[covid_grupoD_infected['date']>'2020-03-15']
total_s

NameError: name 'covid_grupoD_infected' is not defined

In [156]:
total_bar = total_s.groupby(["date","new_date",'iso_code','location'])["location","total_cases","new_cases","total_deaths","new_deaths"].sum().reset_index()
total_bar

NameError: name 'total_s' is not defined

In [157]:
fig = px.bar(total_bar, x="location", y="total_cases", color="location",
              animation_frame="new_date", animation_group="iso_code",  range_y=[0,total_bar.total_cases.max()+1000],title= "Total cases by Country") 
fig.show()

NameError: name 'total_bar' is not defined

-----
Voy a crear un df con los datos sumados de los 5 paises de nuestro estudio por día.
Realizare graficos de barras, lineas, puntos y quesitos del df obtenido anteriormente llamado covid_grupo_infected 

In [158]:
st = covid_grupoD_infected.groupby("date")["new_cases","new_deaths"].sum().reset_index()
st.head(5)

NameError: name 'covid_grupoD_infected' is not defined

In [159]:
# Create with bars the daily deaths and infected in the five countries.
fig = go.Figure(data=[
    go.Bar(name='Daily cases', x=st['date'], y=st['new_cases']),
    go.Bar(name='Daily deaths', x=st['date'], y=st['new_deaths'])])
# Change the bar mode
fig.update_layout(barmode='overlay', title='Daily New Casea and Deaths in ESP, PRT, VEN, GBR, TUR')
fig.show()

NameError: name 'st' is not defined

In [160]:
covid_infected_ESP = covid_grupoD_infected[covid_grupoD_infected['iso_code']== 'ESP']
covid_infected_ESP.head(5)

NameError: name 'covid_grupoD_infected' is not defined

In [161]:
# If I will want create for every country
# Utilizo los df que he creado anteriormente para cada país
fig = go.Figure(data=[
    go.Bar(name='Daily cases', x=covid_infected_ESP['date'], y=covid_infected_ESP['new_cases']),
    go.Bar(name='Daily deaths', x=covid_infected_ESP['date'], y=covid_infected_ESP['new_deaths'])])
# Change the bar mode
fig.update_layout(barmode='overlay', title='Daily Case and Death in ESP')
fig.show()

NameError: name 'covid_infected_ESP' is not defined

Si lo veis interesante se pueden sacar los graficos para los otros cuatro paises.

In [162]:
#Grafico de puntos para nuevos casos diarios
fig = px.scatter(covid_grupoD_infected, x="date", y="new_cases", color="location",
                 facet_col="iso_code", title="Daily new Cases")
fig.update_xaxes(showgrid=False)
fig.show()

NameError: name 'covid_grupoD_infected' is not defined

In [163]:
#Grafico de puntos para nuevos casos diarios
fig = px.scatter(covid_grupoD_infected, x="date", y="new_deaths", color="location",
                 facet_col="iso_code", title="Daily new Deaths")
fig.update_xaxes(showgrid=False)
fig.show()

NameError: name 'covid_grupoD_infected' is not defined

In [164]:
#Grafico de puntos sobre los nuevos casos diarios en España.
fig = px.scatter(covid_infected_ESP, x="date", y="new_cases", color="location",
                 facet_col="iso_code", title="Nuevos casos diarios en España")
fig.update_xaxes(showgrid=False)
fig.show()

NameError: name 'covid_infected_ESP' is not defined

In [165]:
#Grafico de puntos sobre los fallecimientos diarios en España.
fig = px.scatter(covid_infected_ESP, x="date", y="new_deaths", color="location",
                 facet_col="iso_code", title="Fallecimientos diarios en España")
fig.update_xaxes(showgrid=False)
fig.show()

NameError: name 'covid_infected_ESP' is not defined

In [166]:
#Grafico de puntos sobre los nuevos casos diarios en GBR.
fig = px.scatter(covid_grupoD_GBR, x="date", y="new_cases", color="location",
                 facet_col="iso_code", title="Nuevos casos diarios en GBR")
fig.update_xaxes(showgrid=True)
fig.show()

NameError: name 'covid_grupoD_GBR' is not defined

In [167]:
#Grafico de puntos sobre los nuevos fallecimientos diarios en GBR.
fig = px.scatter(covid_grupoD_GBR, x="date", y="new_deaths", color="location",
                 facet_col="iso_code", title="Nuevos fallecimientos diarios en GBR")
fig.update_xaxes(showgrid=True)
fig.show()

NameError: name 'covid_grupoD_GBR' is not defined

Se podria continuar con el resto de los paises

In [168]:
covid_grupoD.head(5)

NameError: name 'covid_grupoD' is not defined

# Are there outliers or some rare data?
See if there are some outliers for the columns:
- new_cases, new_deaths
- total_cases_per_million, new_cases_per_million

(total_cases & total_deaths no creo que tenga mucho interes graficarlos ya que son cantidades acumulativas y aunque en algún mto puntual suban o bajen, como cuando España reviso las cifras, nuncan van a ser outliers. No obstante grafico total_deaths para que se pueda comprobar)

In [169]:
# Outliers de total_deaths
sns.set(style="ticks")
ax = sns.boxplot(x=covid_grupoD["total_deaths"])

NameError: name 'covid_grupoD' is not defined

In [170]:
# Para visualizar los posibles outliers de la columna new_deaths del df de los 5 paises asignados.
sns.set(style="whitegrid")
ax = sns.boxplot(x=covid_grupoD["new_deaths"])

NameError: name 'covid_grupoD' is not defined

In [171]:
# Busqueda de outliers de new_cases
sns.set(style="ticks")
ax = sns.boxplot(x=covid_grupoD["new_cases"])

NameError: name 'covid_grupoD' is not defined

In [172]:
# Busqueda de outliers de new_cases_per_million
sns.set(style="ticks")
ax = sns.boxplot(x=covid_grupoD["new_cases_per_million"]) 

NameError: name 'covid_grupoD' is not defined

In [173]:
# Busqueda de outliers de new_deaths_per_million
sns.set(style="ticks")
ax = sns.boxplot(x=covid_grupoD["new_deaths_per_million"])

NameError: name 'covid_grupoD' is not defined

In [174]:
# Boxplot para mostrar casos totales por pais
sns.set(style="ticks", palette="vlag")
sns.boxplot(x="location", y="total_cases",
             palette=["m", "g","b","r","y"],
            data=covid_grupoD)
sns.despine(offset=10, trim=True)

NameError: name 'covid_grupoD' is not defined

In [175]:
# Boxplot para mostrar muertes totales por pais
sns.set(style="ticks", palette="vlag")
sns.boxplot(x="location", y="total_deaths",
             palette=["m", "g","b","r","y"],
            data=covid_grupoD)
sns.despine(offset=10, trim=True)

NameError: name 'covid_grupoD' is not defined

In [176]:
# Boxplot para mostrar casos diarios por pais
sns.set(style="ticks", palette="vlag")
sns.boxplot(x="location", y="new_cases",
             palette=["m", "g", "b", "r", "y"],
            data=covid_grupoD)
sns.despine(offset=10, trim=True)

NameError: name 'covid_grupoD' is not defined

In [177]:
# Boxplot para mostrar casos diarios por pais mostrando los puntos sobre las cajas
sns.set(style="ticks", palette="vlag")
sns.boxplot(x="location", y="new_cases",
             palette=["m", "g", "b", "r", "y"],
            data=covid_grupoD)
sns.despine(offset=10, trim=True)
ax = sns.swarmplot(x="location", y="new_cases", data=covid_grupoD, color=".25")

NameError: name 'covid_grupoD' is not defined

In [178]:
# Boxplot para mostrar muertes diarias por pais
sns.set(style="ticks", palette="vlag")
sns.boxplot(x="location", y="new_deaths",
             palette=["m", "g","b","r","y"],
            data=covid_grupoD)
sns.despine(offset=10, trim=True)

NameError: name 'covid_grupoD' is not defined

In [179]:
-----
Para representar los casos y fallecidos totales mediante pie charts.

SyntaxError: invalid syntax (<ipython-input-179-07d401dd0a1a>, line 1)

In [180]:
covid_grupoD_total= covid_grupoD_infected.groupby(['iso_code'])['location','total_deaths', 'total_cases'].max()
covid_grupoD_total

NameError: name 'covid_grupoD_infected' is not defined

In [181]:
# Representación de los casos totales en los cinco paises objetos de nuestro estudio
fig = px.pie(covid_grupoD_total, values='total_cases', names='location', title='Casos totales en ESP, VEN, PRT, GBR, TUR')
fig.show()

NameError: name 'covid_grupoD_total' is not defined

In [182]:
# Representación de los fallecidos totales en los cinco paises objetos de nuestro estudio
fig = px.pie(covid_grupoD_total, values='total_deaths', names='location', title='Fallecidos totales en ESP, VEN, PRT, GBR, TUR')
fig.show()

NameError: name 'covid_grupoD_total' is not defined

Para representar los casos y fallecidos diarios creo un nuevo df a partir de covid_grupoD 

In [183]:
covid_grupoD_diario= covid_grupoD_infected.groupby(['date'])['new_deaths', 'new_cases','new_date'].max()
covid_grupoD_diario.head(5)

NameError: name 'covid_grupoD_infected' is not defined

In [184]:
fig = px.pie(covid_grupoD_diario, values='new_deaths', names='new_date', title='Fallecidos diarios en ESP, VEN, PRT, GBR, TUR')
fig.show()
# Muy artistico pero no es util para nosotros

NameError: name 'covid_grupoD_diario' is not defined

-----
Para representar los casos y fallecidos diarios mediante line.

In [185]:
fig = px.line(covid_grupoD_infected, x="date", y="new_cases", color="location", line_group="location", hover_name="location",
        line_shape="spline", render_mode="svg",title='Nuevos casos diarios en ESP, VEN, PRT, GBR, TUR')
fig.show()

NameError: name 'covid_grupoD_infected' is not defined

In [186]:
fig = px.line(covid_grupoD_infected, x="date", y="new_deaths", color="location", line_group="location", hover_name="location",
        line_shape="spline", render_mode="svg",title='Fallecidos diarios en ESP, VEN, PRT, GBR, TUR')
fig.show()

NameError: name 'covid_grupoD_infected' is not defined

# How is the progression going each ten days (of the daily infected rate)?
option A 3.b

desde el df covid_grupoD pondre la columna date como index y lo agrupare

In [187]:
covid_grupoD_date_indice=(covid_grupoD.set_index('date')).groupby(['date','location'])['new_cases','new_deaths'].sum()
covid_grupoD_date_indice

NameError: name 'covid_grupoD' is not defined

In [188]:
covid_grupoD_10days = covid_grupoD_date_indice.resample('10D', level=0).sum().reset_index()
covid_grupoD_10days.head(4)

NameError: name 'covid_grupoD_date_indice' is not defined

In [189]:
# La progresión de los infectados cada 10 dias 
fig = px.line(covid_grupoD_10days, x="date", y="new_cases", title='Daily infected cases')
fig.show()

NameError: name 'covid_grupoD_10days' is not defined

-----
A continuacion voy a tratar de graficar la columna de stringency index en función de otras variables

In [190]:
covid_grupoD.head(5)

NameError: name 'covid_grupoD' is not defined

In [191]:
# para sacar la grafica de strangecy index en función del tiempo
fig = px.line(covid_grupoD, x='date', y='stringency_index', color="iso_code", line_group="iso_code", hover_name="location",
        line_shape="spline", render_mode="svg")
fig.show()

NameError: name 'covid_grupoD' is not defined

In [192]:
#Esto ha sido un intento de grafico de barras que en función del tiempo fuera subiendo, pero no funciona
fig = px.bar(total_s, x="iso_code", y="new_cases",color='stringency_index',
            animation_frame='new_date',animation_group='iso_code', range_y=[0,4000000])
fig.show()

NameError: name 'total_s' is not defined

In [193]:
# Grafico de barras de España comparando medidas restritivas y casos diarios.
fig = go.Figure(data=[
    go.Bar(name='New daily cases', x=stringency_graphic_ESP['date'], y=stringency_graphic_ESP['new_cases_per_million']),
    go.Bar(name='Stringency index', x=stringency_graphic_ESP['date'], y=stringency_graphic_ESP['stringency_index'])])
# Change the bar mode
fig.update_layout(barmode='overlay', title='Daily Case and Daily Death en España')
fig.show()
# En mi opinion no aporta nada y es dificil de interpretar

NameError: name 'stringency_graphic_ESP' is not defined